tests/js/session-tests.js

```diff
@@ -829,20 +830,57 @@ module.exports = {
             });
     },
 
-    testNullPartitionValue() {
-        const app = new Realm.App(appConfig);
-        const credentials = Realm.Credentials.anonymous();
+    async testAcceptedPartitionValueTypes() {
+        const testPartitionValues = [
+            Utils.genPartition(), // string
+            0,
+            Math.floor(Math.random() * Number.MAX_SAFE_INTEGER),
```

Comment:
In general we should avoid random test input. It will make it much harder to reproduce a failing test.

Reply:
This was mostly to not only test the extremes, I'll replace it with a static "random" small number & a large number.

Reply:
..  Or, alternatively, pre-generate a set of 40 random numbers, add the the extremes to the list and test on those..

Reply:
That was the idea, although where you thought 40, I thought 2 - I'll try to find something mid way.